File: examples/t-0-0.js
#!/usr/bin/env nodejs
var config = {
    user: 'sa',
    password: 'SaAdmin1@',
    server: '127.0.0.1',
    database: 'testing',

    options: {
        encrypt: false // Use this if you're on Windows Azure
    }
};
const stmt = "select idrow, data from t where data = 'some %% value'";

const sql = require('mssql')

sql.connect(config, err => {
    const request = new sql.Request();
    request.stream = true // You can set streaming differently for each request
    console.log('execute', stmt);
    request.query(stmt);

    request.on('recordset', columns => {
        console.log("recordset", columns);
    })

    request.on('row', row => {
        console.log("row", row);
    })

    request.on('error', err => {
        console.log("error", err);
    })

    request.on('done', result => {
        console.log("done", result);
        process.exit(0);
    })

});

sql.on('error', err => {
    console.log("error", err)
})

File: examples/t-0-0.py
#!/usr/bin/env vpython3
from t import username, userpass
import pytds


class Main(object):
    def __init__(self):
        self.conn = pytds.connect(
            dsn="127.0.0.1", user=username, password=userpass, database="testing"
        )

    def close(self):
        self.conn.close()

    def bug1(self):
        print("*" * 20, "bug 0")
        sqls = (
            "create table #t(idrow int identity(1, 1) primary key,data varchar(50))",
        )
        c = self.conn.cursor()
        for sql in sqls:
            c.execute(sql)
        c.close()
        # self.conn.commit()

        sql = "INSERT INTO #t (data) VALUES (%(data)s)"
        c = self.conn.cursor()
        c.execute(sql, {"data": "some % value"})
        c.execute(sql, {"data": "some %% value"})
        c.close()
        # self.conn.commit()

        sqls = (
            """SELECT data FROM #t""",
            """SELECT data FROM #t WHERE data = 'some % value' """,
            """SELECT data FROM #t WHERE data = 'some %% value' """,
        )
        c = self.conn.cursor()
        for sql in sqls:
            print("*" * 20)
            print(sql)
            c.execute(sql)
            for rs in c.fetchall():
                print(rs)
        c.close()

    def main(self):
        self.bug1()


def main():
    cls = Main()
    try:
        cls.main()
    finally:
        cls.close()


main()
